viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/process/strategy/DefaultSectionCalculatorStrategy.scala

```diff
@@ -129,25 +129,58 @@ class DefaultSectionCalculatorStrategy extends RoadAddressSectionCalculatorStrat
   }
 
   private def continuousRoadwaySection(seq: Seq[ProjectLink], givenRoadwayNumber: Long): (Seq[ProjectLink], Seq[ProjectLink]) = {
-    val track =
-      seq.headOption.map(_.track).getOrElse(Track.Unknown)
-    val discontinuity =
-      seq.head.discontinuity
-    val administrativeClass =
-      seq.head.administrativeClass
-    val continuousProjectLinks =
-      seq.takeWhile(pl => pl.track == track && pl.discontinuity == discontinuity && pl.administrativeClass == administrativeClass)
+    def createConnectedSection(startLink: ProjectLink, allLinks: Seq[ProjectLink]): List[ProjectLink] = {
```

Comment:
This assumes address value ordered ProjectLinks?
That would be good to say in function comments.

Reply:
Hmm, create points to database write at least in my mind...
pickConnectedSection instead? 

And!
In fact, picks stuff geametrically. So.... pickGeometricallyConnectedSection?